Fix `unit.modules.test_cmdmod` for Windows

twangboy opened the pull request:
Fixes the `unit.modules.test_cmdmod` test on Windows.

salt.utils.win_runas
- Returns CommandExecutionError when the users does not exist
- Fixes some spelling

tests.unit.modules.test_cmdmod
- Fixes some linesep issues
- Combines some mocking

### Tests written?
Yes

### Commits signed with GPG?
Yes

twangboy committed:
Fix `unit.modules.test_cmdmod` for Windows

Fixes the `unit.modules.test_cmdmod` test on Windows.

salt.utils.win_runas
- Returns CommandExecutionError when the users does not exist
- Fixes some spelling

tests.unit.modules.test_cmdmod
- Fixes some linesep issues
- Combines some mocking

rallytime reviewed (approved):
👍

rallytime commented:
@twangboy Looks like one of the tests is still failing on windows:

https://jenkinsci.saltstack.com/job/pr-kitchen-windows2016-py2/job/PR-49953/1/testReport/junit/unit.modules.test_cmdmod/CMDMODTestCase/test_run_all_binary_replace/

And is also failing on linux:

https://jenkinsci.saltstack.com/job/pr-kitchen-ubuntu1604-py3/job/PR-49953/1/

twangboy committed:
Fix failing test

In py3 bytes are actual bytes and you can't join them with strings
So, you have to make sure everything's the same type
I used `.encode()`
